xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp

```diff
@@ -268,15 +268,20 @@ bool CDVDInputStreamPVRManager::PrevChannel(bool preview/* = false*/)
 bool CDVDInputStreamPVRManager::SelectChannelByNumber(unsigned int iChannelNumber)
 {
   PVR_CLIENT client;
+  CPVRChannelPtr currentChannel(g_PVRManager.GetCurrentChannel());
+  CFileItemPtr item(g_PVRChannelGroups->Get(currentChannel->IsRadio())->GetSelectedGroup()->GetByChannelNumber(iChannelNumber));
```

Comment:
I'm not sure why this is different from `CPVRManager::GetPlayingGroup(bool bRadio)`.

The call hierarchy:
`g_PVRManager.GetPlayingGroup(bool bRadio)`
`CPVRChannelGroupsContainer::GetSelectedGroup(bool bRadio)`
`CPVRChannelGroupsContainer::Get(bool bRadio)`
`CPVRChannelGroups::GetSelectedGroup(void)`

Your call has the following call hierarchy. 
`g_PVRChannelGroups`
`g_PVRManager.ChannelGroups()`
`CPVRChannelGroupsContainer::Get(bool bRadio)`
`CPVRChannelGroups::GetSelectedGroup(void)`